checksum.go

```
@@ -316,6 +317,67 @@ func (c ChecksumType) String() string {
+	n, err := c.ReadCloser.Read(p)
+	if n > 0 {
+		n2, err := c.Hash.Write(p[:n])
+		if err != nil {
+			return 0, err
+		}
+		if n2 != n {
+			return 0, io.ErrShortWrite
+		}
+	}
+	if err == io.EOF {
+		if got := base64.StdEncoding.EncodeToString(c.Hash.Sum(nil)); got != c.expectChecksum {
```

golangci-lint (2.12.2)

[error] QF1008: could remove embedded field "Hash" from selector

(staticcheck)